electrum/gui/qt/transaction_dialog.py

```diff
@@ -245,6 +251,9 @@ def insert_tx_io(
                 tcf_shortid=tcf_shortid,
                 short_id=str(txin.short_id), addr=addr, value=txin_value,
             )
+            if isinstance(txin, PartialTxInput):
+                if txin.sighash != Sighash.ALL and self.wallet.adb.is_mine(addr):
```

Comment:
A related but distinct point re adding warnings in the GUIs is what to do about the CLI. E.g. consider calling the `signtransaction` command with a PSBT given to us by someone else (e.g. coinjoin). Perhaps we should just hard fail if the sighashes are not sighash_all, by default.

This can be handled separately and does not have to be solved in this PR though.

Reply:
Yeah we should give this some thought, see how we can share as much non-gui specific gui code in shared code (separate layer?)

Reply:
> A related but distinct point re adding warnings in the GUIs is what to do about the CLI. E.g. consider calling the `signtransaction` command with a PSBT given to us by someone else (e.g. coinjoin). Perhaps we should just hard fail if the sighashes are not sighash_all, by default.

Added the checks to `wallet.sign_transaction()` with a kwarg to not consider warnings as fatal, which can be set through the `--iknowwhatimdoing` cmdline arg